tedana/decomposition/eigendecomp.py

```diff
@@ -51,16 +51,123 @@ def run_svd(data):
         Component timeseries.
     """
     # do PC dimension selection and get eigenvalue cutoff
-    ppca = PCA(n_components='mle', svd_solver='full')
+    if mle:
+        ppca = PCA(n_components='mle', svd_solver='full')
+    else:
+        ppca = PCA()
     ppca.fit(data)
     v = ppca.components_
     s = ppca.explained_variance_
     u = np.dot(np.dot(data, v.T), np.diag(1. / s))
     return u, s, v
 
 
-def tedpca(catd, OCcatd, combmode, mask, t2s, t2sG, stabilize,
-           ref_img, tes, kdaw, rdaw, ste=0, wvpca=False):
+def kundu_tedpca(ct_df, n_echos, kdaw, rdaw, stabilize=False):
```

Comment:
I think we should move this into `selection`, given that it doesn't actually run PCA and is just used for component selection. Thoughts?

Reply:
Strongly agree ! :+1:

Reply:
I spent some time figuring out how best to do this, and I think I'd like to push that to a separate PR actually. There are some elements of tedpca that I think we should discuss in more detail before we can reorganize it to better mirror tedica. Just as an example, I'm not sure why we use an `eimask`-generated mask when computing Kappa and Rho values for tedpca, but not for tedica.

Reply:
That would be great, thanks!

Reply:
Created: https://github.com/ME-ICA/tedana/issues/166